common_types/src/datum.rs

```diff
@@ -1085,4 +1243,101 @@ mod tests {
             assert!(source.to_negative().is_none());
         }
     }
+
+    #[test]
+    fn test_parse_datum_date_from_str() {
+        let cases = ["-9999-01-01", "9999-12-21", "0-1-1", "1000-02-28"];
+
+        let expects = [
+            Datum::Date(-4371587),
+            Datum::Date(2932886),
+            Datum::Date(-719528),
+            Datum::Date(-354227),
+        ];
```

Comment:
Actually, with the `parse_datum_date_from_str` and `format_datum_date`, there is no need to define the expects, and what we should test is the condition: `v == format_datum_date(parse_datum_date_from_str(v))`.

Reply:
done